web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java

```diff
@@ -65,6 +67,9 @@ static AuthTokenPromiseSupplier oneShot(ConnectToken initialToken) {
     @JsProperty(namespace = "dh.QueryInfo")
     public static final String EVENT_CONNECT = "connect";
 
+    @JsProperty(namespace = "dh.IdeConnection")
+    public static final String HACK_CONNECTION_FAILURE = "hack-connection-failure";
```

Comment:
Why is this named `hack-connection-failure`? Why `hack-`?

Reply:
I believe 'hack' here is to make it blindly obvious this is a temporary work around to a bigger problem that Colin wants carte blanch to fix correctly.

Reply:
This is temporary, until we have reconnect (#730) which will have this being removed. I wanted to draw a distinction between normal events or internal-only ones, but I'm open to some other way to do that.

I tried to summarize in the description above (and will make it part of the actual commit message), but the old reconnect events are probably what we will use for these cases once the api is smart enough to try reconnecting streams.

Reply:
Okay cool, I've got a change to listen for these new errors pushed on my Application Mode PR: https://github.com/deephaven/web-client-ui/pull/155